Source/RapyutaSimulationPlugins/Public/Core/RRBaseActor.h

```diff
@@ -112,6 +112,23 @@ class RAPYUTASIMULATIONPLUGINS_API ARRBaseActor : public APawn
     FTimerHandle GenericTimerHandle;
 
 public:
+    /**
+     * @brief ACTOR INTIALIZING GENERAL INFO (Unique name, mesh list, material list, etc.)
+     *
+     * @tparam TActorSpawnInfo
+     * @param InActorInfo
+     * @return true
+     * @return false
+     */
+    template<typename TActorSpawnInfo>
```

Comment:
- Do we need to make this template?
- Can we make this merge to Intiialize()? i.e.
```
bool Initialize(const TActorSpawnInfo& InActorInfo = nullptr)
{
 if(InActorInfo){ 
    ActorInfo=InActorInfo
}
...

Reply:
So the template's purpose is to allow users to use a custom ActorSpawnInfo struct inheriting from `FRRActorSpawnInfo` and still be able to instantiate it.
Right, merging them into one makes sense.

Reply:
So the template's purpose is to allow users to use a custom ActorSpawnInfo struct inheriting from `FRRActorSpawnInfo` and still be able to instantiate it.
Ah I forgot, since a template cannot be made `virtual`, which is required here so yeah we could not merge those for that reason.